mifosng-android/src/main/java/com/mifos/api/datamanager/DataManagerSearch.java

```diff
@@ -17,14 +25,55 @@
 public class DataManagerSearch {
 
     public final BaseApiManager baseApiManager;
+    public final DatabaseHelperClient mDatabaseHelperClient;
+    public final DatabaseHelperGroups mDatabaseHelperGroup;
+    public final DatabaseHelperCenter mDatabaseHelperCenter;
+    public final DatabaseHelperLoan mDatabaseHelperLoan;
+    public final DatabaseHelperSavings mDatabaseHelperSaving;
 
     @Inject
-    public DataManagerSearch(BaseApiManager baseApiManager) {
+    public DataManagerSearch(BaseApiManager baseApiManager,
+                             DatabaseHelperClient databaseHelperClient,
+                             DatabaseHelperGroups databaseHelperGroup,
+                             DatabaseHelperCenter databaseHelperCenter,
+                             DatabaseHelperLoan databaseHelperLoan,
+                             DatabaseHelperSavings databaseHelperSaving) {
         this.baseApiManager = baseApiManager;
+        mDatabaseHelperClient = databaseHelperClient;
+        mDatabaseHelperGroup = databaseHelperGroup;
+        mDatabaseHelperCenter = databaseHelperCenter;
+        mDatabaseHelperLoan = databaseHelperLoan;
+        mDatabaseHelperSaving = databaseHelperSaving;
     }
 
-    public Observable<List<SearchedEntity>> searchResources(String query, String resources,
-                                                            Boolean exactMatch) {
-        return baseApiManager.getSearchApi().searchResources(query, resources, exactMatch);
+    public Observable<List<SearchedEntity>> searchResources(final String query, String resources,
+                                                            final Boolean exactMatch) {
+        switch (PrefManager.getUserStatus()) {
+            case 0:
+                return baseApiManager.getSearchApi().searchResources(query, resources, exactMatch);
+            case 1:
+                if (resources.equals(Constants.ENTITY_TYPE_CLIENTS)) {
+                    return mDatabaseHelperClient.readAllClientsByQuery(query, exactMatch);
+                } else if (resources.equals(Constants.ENTITY_TYPE_GROUPS)) {
+                    return
+                            mDatabaseHelperGroup.readAllGroupsByQuery(query, exactMatch)
+                                    .mergeWith(mDatabaseHelperCenter.readAllCentersByQuery(query,
```

Comment:
@therajanmaurya It is giving me same error as my implementation. However, when I am commenting line 55 and 56 in onError function - 
https://github.com/openMF/android-client/blob/master/mifosng-android/src/main/java/com/mifos/mifosxdroid/online/search/SearchPresenter.java#L55
The error is gone. But even after that, it is showing results from only one observable.
I have done this - 
return Observable.merge(mDatabaseHelperGroup.readAllGroupsByQuery(query, exactMatch), mDatabaseHelperCenter.readAllCentersByQuery(query, exactMatch));